lib/commands/publish-ios.ts

```diff
@@ -0,0 +1,46 @@
+///<reference path="../.d.ts"/>
+"use strict";
+
+import {StringCommandParameter} from "../common/command-params";
+
+export class PublishIOS implements ICommand {
+	constructor(private $injector: IInjector,
+		private $itmsTransporterService: IITMSTransporterService,
+		private $logger: ILogger,
+		private $options: IOptions,
+		private $prompter: IPrompter,
+		private $stringParameterBuilder: IStringParameterBuilder) { }
+
+	public allowedParameters: ICommandParameter[] = [this.$stringParameterBuilder.createMandatoryParameter("Missing required Apple ID parameter."),
+		new StringCommandParameter(this.$injector), new StringCommandParameter(this.$injector), new StringCommandParameter(this.$injector), new StringCommandParameter(this.$injector)];
+
+	public execute(args: string[]): IFuture<void> {
+		return (() => {
+			let app_id = args[0],
+				userName = args[1],
+				password = args[2],
+				mobileProvisionIdentifier = args[3],
+				codeSignIdentity = args[4];
+
+			if(!userName) {
+				userName = this.$prompter.getString("Apple ID", { allowEmpty: false }).wait();
+			}
+
+			if(!password) {
+				password = this.$prompter.getPassword("Apple ID password").wait();
+			}
+
+			if(!mobileProvisionIdentifier) {
+				this.$logger.warn("Mobile Provision identifier not set - a default one will be used. You can set one in App_Resources/iOS/build.xcconfig");
```

Comment:
this message may be confusing in case `build.xcconfig` does not exist (that's the default case). Maybe we should consider adding default `build.xcconfig` in `app/App_Resources/iOS`

Also please add `app` in front of `App_Resources`, so the user will be sure where to place the file.